cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java

```diff
@@ -165,9 +173,13 @@ public void execute() {
                 }
                 CentralAPIClient client = new CentralAPIClient(RepoUtils.getRemoteRepoURL(),
                         initializeProxy(settings.getProxy()), getAccessTokenOfCLI(settings));
-
                 try {
-                    pushPackage(project, client);
+                    if (balaPath == null) {
+                        pushPackage(project, client);
+                    } else {
+                        // Skip validation entirely and push to central
+                        pushBalaToRemote(balaPath, client);
```

Comment:
We shouldn't ignore all the validations. We can still validate the extension and the files inside the bala.

Reply:
Added the file validation